<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>XVideoUIClass</class>
 <widget class="QWidget" name="XVideoUIClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>846</width>
    <height>593</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>XVideoUI</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QPushButton:hover
 { 
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1, stop:0 rgba(120, 120, 120, 255), stop:0.495 rgba(39, 39, 39, 255), stop:0.505 rgba(39,39, 39, 255), stop:1 rgba(120, 120,120, 255));
 border: none;
border-radius:5px;
color: rgb(255, 255, 255);
font: 75 12pt &quot;黑体&quot;;
 }
QPushButton:!hover
 { 
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1, stop:0 rgba(39, 39, 39, 255), stop:0.495 rgba(120, 120, 120, 255), stop:0.505 rgba(120,120, 120, 255), stop:1 rgba(39, 39,39, 255));
 border: none;
border-radius:5px;
color: rgb(255, 255, 255);
font: 75 12pt &quot;黑体&quot;;
 }
#XVideoUIClass
{
	background-color: rgb(53, 53, 53);
}
#closebutton
{
font: 75 11pt &quot;等线&quot;;
color: rgb(255, 255, 255);
background-color: rgb(53, 53, 53);
}

#playButton{background-color: rgba(255, 255, 255, 0);}

#playButton:!hover{ image: url(:/XVideoUI/Resources/PlayN.png);}
#playButton:hover{ image: url(:/XVideoUI/Resources/PlayH.png);}

#playButton:pressed{ image: url(:/XVideoUI/Resources/PlayP.png);}

#pauseButton{background-color: rgba(255, 255, 255, 0);}

#pauseButton:!hover{ image: url(:/XVideoUI/Resources/PauseN.png);}
#pauseButton:hover{ image: url(:/XVideoUI/Resources/PauseH.png);}

#pauseButton:pressed{ image: url(:/XVideoUI/Resources/PauseP.png);}


QLabel
{
	
	color: rgb(255, 255, 255);
	
	font: 75 12pt &quot;黑体&quot;;
}
</string>
  </property>
  <widget class="XVideoWidget" name="src1">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>30</y>
     <width>401</width>
     <height>251</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="openfile">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>300</y>
     <width>71</width>
     <height>41</height>
    </rect>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="text">
    <string>打开</string>
   </property>
  </widget>
  <widget class="QPushButton" name="closebutton">
   <property name="geometry">
    <rect>
     <x>810</x>
     <y>10</y>
     <width>31</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>X</string>
   </property>
   <property name="flat">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QSlider" name="playSlider">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>300</y>
     <width>401</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="XVideoWidget" name="des">
   <property name="geometry">
    <rect>
     <x>420</x>
     <y>30</y>
     <width>401</width>
     <height>251</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>410</x>
     <y>350</y>
     <width>91</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>亮度[0-100]</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="bright">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>350</y>
     <width>151</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="label_2">
   <property name="geometry">
    <rect>
     <x>380</x>
     <y>380</y>
     <width>121</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>对比度[1.0-3.0]</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="contrast">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>380</y>
     <width>151</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>1.000000000000000</double>
   </property>
   <property name="maximum">
    <double>3.000000000000000</double>
   </property>
  </widget>
  <widget class="QPushButton" name="setButton">
   <property name="geometry">
    <rect>
     <x>710</x>
     <y>360</y>
     <width>71</width>
     <height>41</height>
    </rect>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="text">
    <string>设置</string>
   </property>
  </widget>
  <widget class="QPushButton" name="exportButton">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>300</y>
     <width>71</width>
     <height>41</height>
    </rect>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="text">
    <string>导出</string>
   </property>
  </widget>
  <widget class="QPushButton" name="playButton">
   <property name="geometry">
    <rect>
     <x>130</x>
     <y>330</y>
     <width>41</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QPushButton" name="pauseButton">
   <property name="geometry">
    <rect>
     <x>180</x>
     <y>330</y>
     <width>41</width>
     <height>41</height>
    </rect>
   </property>
   <property name="text">
    <string/>
   </property>
  </widget>
  <widget class="QLabel" name="label_3">
   <property name="geometry">
    <rect>
     <x>430</x>
     <y>410</y>
     <width>71</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>图形旋转</string>
   </property>
  </widget>
  <widget class="QComboBox" name="rotate">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>410</y>
     <width>151</width>
     <height>22</height>
    </rect>
   </property>
   <property name="currentText">
    <string>0</string>
   </property>
   <property name="currentIndex">
    <number>0</number>
   </property>
   <item>
    <property name="text">
     <string>0</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>90</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>180</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>270</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label_4">
   <property name="geometry">
    <rect>
     <x>430</x>
     <y>440</y>
     <width>71</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>图形镜像</string>
   </property>
  </widget>
  <widget class="QComboBox" name="flip">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>440</y>
     <width>151</width>
     <height>22</height>
    </rect>
   </property>
   <property name="editable">
    <bool>false</bool>
   </property>
   <property name="currentText">
    <string>不处理</string>
   </property>
   <property name="currentIndex">
    <number>0</number>
   </property>
   <item>
    <property name="text">
     <string>不处理</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>上下镜像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>左右镜像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>上下左右镜像</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="label_5">
   <property name="geometry">
    <rect>
     <x>400</x>
     <y>470</y>
     <width>101</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>图像尺寸(W.H)</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="width">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>470</y>
     <width>81</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>9999</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="height">
   <property name="geometry">
    <rect>
     <x>600</x>
     <y>470</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>9999</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_6">
   <property name="geometry">
    <rect>
     <x>280</x>
     <y>500</y>
     <width>221</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>图像金字塔（高斯、拉普拉斯）</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="pyup">
   <property name="geometry">
    <rect>
     <x>600</x>
     <y>500</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="pydown">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>500</y>
     <width>81</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
  </widget>
  <widget class="QLabel" name="label_7">
   <property name="geometry">
    <rect>
     <x>280</x>
     <y>530</y>
     <width>221</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>图像裁剪（x,y,width,height）</string>
   </property>
  </widget>
  <widget class="QSpinBox" name="cy">
   <property name="geometry">
    <rect>
     <x>600</x>
     <y>540</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="cx">
   <property name="geometry">
    <rect>
     <x>520</x>
     <y>540</y>
     <width>81</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="ch">
   <property name="geometry">
    <rect>
     <x>750</x>
     <y>540</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="cw">
   <property name="geometry">
    <rect>
     <x>670</x>
     <y>540</y>
     <width>81</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>100</number>
   </property>
  </widget>
  <widget class="QComboBox" name="color">
   <property name="geometry">
    <rect>
     <x>710</x>
     <y>430</y>
     <width>91</width>
     <height>22</height>
    </rect>
   </property>
   <item>
    <property name="text">
     <string>RGB图像</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>灰度图</string>
    </property>
   </item>
  </widget>
  <widget class="QSpinBox" name="mx">
   <property name="geometry">
    <rect>
     <x>100</x>
     <y>380</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>9999</number>
   </property>
  </widget>
  <widget class="QSpinBox" name="my">
   <property name="geometry">
    <rect>
     <x>170</x>
     <y>380</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="maximum">
    <number>9999</number>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="ma">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>380</y>
     <width>71</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1.000000000000000</double>
   </property>
  </widget>
  <widget class="QPushButton" name="markButton">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>370</y>
     <width>81</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>水印x,y,a</string>
   </property>
  </widget>
  <widget class="XVideoWidget" name="src2">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>410</y>
     <width>240</width>
     <height>180</height>
    </rect>
   </property>
  </widget>
  <widget class="QPushButton" name="blendButton">
   <property name="geometry">
    <rect>
     <x>260</x>
     <y>420</y>
     <width>51</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>融合</string>
   </property>
  </widget>
  <widget class="QDoubleSpinBox" name="ba">
   <property name="geometry">
    <rect>
     <x>320</x>
     <y>420</y>
     <width>61</width>
     <height>22</height>
    </rect>
   </property>
   <property name="minimum">
    <double>0.000000000000000</double>
   </property>
   <property name="maximum">
    <double>1.000000000000000</double>
   </property>
   <property name="value">
    <double>0.500000000000000</double>
   </property>
  </widget>
  <widget class="QPushButton" name="mergeButton">
   <property name="geometry">
    <rect>
     <x>260</x>
     <y>460</y>
     <width>51</width>
     <height>31</height>
    </rect>
   </property>
   <property name="text">
    <string>合并</string>
   </property>
  </widget>
  <widget class="QSlider" name="left">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>280</y>
     <width>401</width>
     <height>16</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QSlider::handle:horizontal {
background:  rgb(0, 170, 255);
border-radius: 5px;
}
</string>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
  <widget class="QSlider" name="right">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>320</y>
     <width>401</width>
     <height>16</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">QSlider::handle:horizontal {
background:  rgb(0, 170, 255);
border-radius: 5px;
}
</string>
   </property>
   <property name="maximum">
    <number>999</number>
   </property>
   <property name="value">
    <number>999</number>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>XVideoWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>xvideowidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="XVideoUI.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>openfile</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>433</x>
     <y>329</y>
    </hint>
    <hint type="destinationlabel">
     <x>416</x>
     <y>176</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlider</sender>
   <signal>sliderPressed()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>SliderPress()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>8</x>
     <y>314</y>
    </hint>
    <hint type="destinationlabel">
     <x>25</x>
     <y>321</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlider</sender>
   <signal>sliderReleased()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>SliderRelease()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>389</x>
     <y>314</y>
    </hint>
    <hint type="destinationlabel">
     <x>395</x>
     <y>320</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playSlider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>SetPos(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>110</x>
     <y>321</y>
    </hint>
    <hint type="destinationlabel">
     <x>136</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>closebutton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>826</x>
     <y>22</y>
    </hint>
    <hint type="destinationlabel">
     <x>753</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Set()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>746</x>
     <y>385</y>
    </hint>
    <hint type="destinationlabel">
     <x>762</x>
     <y>371</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>exportButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Export()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>572</x>
     <y>317</y>
    </hint>
    <hint type="destinationlabel">
     <x>589</x>
     <y>444</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>playButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Play()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>142</x>
     <y>350</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>407</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pauseButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Pause()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>207</x>
     <y>360</y>
    </hint>
    <hint type="destinationlabel">
     <x>219</x>
     <y>398</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>markButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Mark()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>78</x>
     <y>386</y>
    </hint>
    <hint type="destinationlabel">
     <x>291</x>
     <y>571</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blendButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Blend()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>284</x>
     <y>437</y>
    </hint>
    <hint type="destinationlabel">
     <x>382</x>
     <y>455</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mergeButton</sender>
   <signal>clicked()</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Merge()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>298</x>
     <y>477</y>
    </hint>
    <hint type="destinationlabel">
     <x>344</x>
     <y>476</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>left</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>Left(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>56</x>
     <y>288</y>
    </hint>
    <hint type="destinationlabel">
     <x>312</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>right</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>XVideoUIClass</receiver>
   <slot>right(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>273</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>453</x>
     <y>2</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>Open()</slot>
  <slot>SliderPress()</slot>
  <slot>SliderRelease()</slot>
  <slot>SetPos(int)</slot>
  <slot>Close()</slot>
  <slot>Set()</slot>
  <slot>Export()</slot>
  <slot>Play()</slot>
  <slot>Pause()</slot>
  <slot>Mark()</slot>
  <slot>Blend()</slot>
  <slot>Merge()</slot>
  <slot>Left(int)</slot>
  <slot>right(int)</slot>
 </slots>
</ui>
